### PR TITLE
Add the ability to parse META.yml files

### PR DESCRIPTION
**Pull request by jcscottiii**
Adds SchemaValue which is a container of helper functions to parse
different types of values for a given key - inspired by quicktype

Add validate_dict to ensure only the given keys in a dictionary are
allowed.

This functionality will be useful for other YAML files in the future.

Notable changes for the reviewer:
- The addition of a yaml folder (which uses the same pyyaml version in other places). This keeps the dependency of yaml to just this folder.

This is part of a multi PR plan
- PR 1 - Parse META.yml files (This PR) - #43394
- PR 2 - Add lint of META.yml files - #43395 
- PR 3 - Parse WEB_FEATURES.yml files - #43403
- PR 4 - Add lint of WEB_FEATURES.yml files - #43401

### CODE REVIEW

File: tools/metadata/meta/schema.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+"""
+YAML filename for meta files
+"""
+META_YML_FILENAME = "META.yml"
+
+@dataclass
+class MetaFile():
+    """documented structure of META files.
+    Reference: https://github.com/web-platform-tests/wpt/pull/18434
```
Comment (foolip) [inline tools/metadata/meta/schema.py:14]: Hmm, it's unfortunate the documentation of `META.yml` is spread across two files. But I agree this PR is then the best thing to point to.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
@jcscottiii can you check out the infrastructure failures?

**Comment by jcscottiii** [comment]
> @jcscottiii can you check out the infrastructure failures?

@foolip Looks like there's a non related issue to this PR. Another PR is having the same issue. https://github.com/web-platform-tests/wpt/pull/43546

Let me see if I can find the faulty code.

**Comment by jcscottiii** [comment]
Looks like the infra issues have been fixed. (More details in this comment: https://github.com/web-platform-tests/wpt/pull/43424#issuecomment-1858184100)

**Pull request merged**